### PR TITLE
Remove image from annual reports

### PR DESCRIPTION
**Pull request by nlisgo**
Improves: https://github.com/elifesciences/jira-import/issues/4192

### CODE REVIEW

File: src/bootstrap.php [hunk 1]
```diff
@@ -454,6 +454,9 @@
     }
 
     foreach ($reports as $i => $report) {
+        if ($type->getParameter('version') < 2 && 2014 === $report['year']) {
```
Comment (thewilkybarkid) [inline src/bootstrap.php]: It'd be better to test for the presence of `$report['image']`, but instead let's keep the data, and instead `unset()` the image when requesting version 2. We can remove the image when we drop version 1.

**Comment by thewilkybarkid** [comment]
Depends on #140 to pass.

File: src/bootstrap.php [hunk 2]
```diff
@@ -454,7 +454,11 @@
     }
 
     foreach ($reports as $i => $report) {
-        unset($report['content']);
```
Comment (nlisgo) [inline src/bootstrap.php]: `content` isn't present anyway in any of the annual report data.

**Comment by nlisgo** [comment]
@giorgiosironi this is green now #140 has made it in. You might want to review the latest commits which address feedback from @thewilkybarkid.

**Comment by giorgiosironi** [comment]
Can look at the changes, but I cannot approve someone else's review (both conceptually and physically)

File: src/bootstrap.php [hunk 3]
```diff
@@ -459,7 +459,11 @@
...
+        if ($type->getParameter('version') < 2 && empty($report['image'])) {
+            throw new NotAcceptableHttpException('At least one of the annual reports in the list requires version 2.');
```
Comment (thewilkybarkid) [inline src/bootstrap.php]: This will always break, so keep the image and remove this.

**Comment by nlisgo** [comment]
@thewilkybarkid I've added the image back in for all `annual-report` fixtures.

File: test/SmokeTest.php [hunk 4]
```diff
@@ -72,18 +72,28 @@ public function requestProvider() : Traversable
...
+            $this->createRequest('/annual-reports/2012', 'application/vnd.elife.annual-report+json; version=3'),
             'application/problem+json',
             406,
         ];
 
         yield $path = '/annual-reports' => [
             $this->createRequest($path),
+            'application/vnd.elife.annual-report-list+json; version=2',
+        ];
+        yield $path = '/annual-reports' => [
```
Comment (thewilkybarkid) [inline test/SmokeTest.php]: We need unique keys here, generators can produce duplicates but not sure how this gets used (so one of these could be missed).

(Hence others having `version 1` etc suffixes.)

Comment (nlisgo) [reply test/SmokeTest.php]: good spot. I've amended the keys.

**Pull request merged**